compiler/lib/src/main/scala/analysis/Semantics/Event.scala

```diff
@@ -35,14 +45,59 @@ object Event {
         )
         else Right(())
       }
+      def getEveryIntervalValue(every: AstNode[Ast.Expr]) = {
+        val loc = Locations.get(every.id)
+        val Value.AnonStruct(intervalValue) = Analysis.convertValueToType(a.valueMap(every.id), Type.AnonStruct(
+          Map(
+            ("seconds", Type.U32),
+            ("useconds", Type.U32),
+          )
+        ))
+        def getNonZeroMember(member: String, maxValue: Option[Int] = None) = {
+          val Value.PrimitiveInt(v, Type.PrimitiveInt.U32) = Analysis.convertValueToType(
+            intervalValue.get(member).get,
+            Type.U32
+          )
+
+          if (v < 0) {
+            Left(SemanticError.InvalidIntValue(
+              loc, v, s"$member may not be negative"
+            ))
+          } else {
+            maxValue match {
```

Comment:
I think we should always have a max value here. For the seconds value, it has to fit in a U32.